dtfs-central-api/src/v1/validation/utilisation-report-service/fee-record-validator.test.ts

```diff
@@ -0,0 +1,174 @@
+      });
+
+      describe('and there are missing fee record ids from the fee record payment group', () => {
+        it('should throw an invalid payload error', async () => {
+          // Arrange
+          const payments = [
+            PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP)
+              .withFeeRecords([
+                FeeRecordEntityMockBuilder.forReport(aReconciliationInProgressReport()).withId(1).build(),
+                FeeRecordEntityMockBuilder.forReport(aReconciliationInProgressReport()).withId(2).build(),
+              ])
+              .build(),
+          ];
+          const feeRecords = [
+            FeeRecordEntityMockBuilder.forReport(aReconciliationInProgressReport()).withId(1).withStatus(status).withPayments(payments).build(),
+          ];
+          feeRecordFindBySpy.mockResolvedValue(feeRecords);
+
+          const feeRecordIds = [1];
+
+          // Act + Assert
+          await expect(validateFeeRecordsFormCompleteGroup(feeRecordIds)).rejects.toThrow(InvalidPayloadError);
+        });
+      });
+    });
+
+    describe('when the fee record statuses do not match', () => {
+      it('should throw an invalid payload error', async () => {
+        // Arrange
+        const feeRecords = [
+          FeeRecordEntityMockBuilder.forReport(aReconciliationInProgressReport()).withId(1).withStatus(FEE_RECORD_STATUS.TO_DO).build(),
+          FeeRecordEntityMockBuilder.forReport(aReconciliationInProgressReport()).withId(2).withStatus(FEE_RECORD_STATUS.DOES_NOT_MATCH).build(),
+          FeeRecordEntityMockBuilder.forReport(aReconciliationInProgressReport()).withId(3).withStatus(FEE_RECORD_STATUS.MATCH).build(),
+        ];
+        feeRecordFindBySpy.mockResolvedValue(feeRecords);
+
+        const feeRecordIds = [1, 2, 3];
+
+        // Act / Assert
+        await expect(validateFeeRecordsFormCompleteGroup(feeRecordIds)).rejects.toThrow(InvalidPayloadError);
+      });
+    });
+
+    describe('when all but one of the fee record statuses do not match', () => {
+      it('should throw an invalid payload error', async () => {
+        // Arrange
+        const feeRecords = [
+          FeeRecordEntityMockBuilder.forReport(aReconciliationInProgressReport()).withId(1).withStatus(FEE_RECORD_STATUS.TO_DO).build(),
+          FeeRecordEntityMockBuilder.forReport(aReconciliationInProgressReport()).withId(2).withStatus(FEE_RECORD_STATUS.DOES_NOT_MATCH).build(),
+          FeeRecordEntityMockBuilder.forReport(aReconciliationInProgressReport()).withId(3).withStatus(FEE_RECORD_STATUS.TO_DO).build(),
+        ];
+        feeRecordFindBySpy.mockResolvedValue(feeRecords);
+
+        const feeRecordIds = [1, 2, 3];
+
+        // Act / Assert
+        await expect(validateFeeRecordsFormCompleteGroup(feeRecordIds)).rejects.toThrow(InvalidPayloadError);
+      });
+    });
+  });
+
+  function aReconciliationInProgressReport() {
+    return UtilisationReportEntityMockBuilder.forStatus('RECONCILIATION_IN_PROGRESS').build();
+  }
```

Comment:
Is it important that we have a report that is in progress? If not would be nice to replace usages of this with `new UtilisationReportEntityMockBuilder().build()` as we have that available to us now!